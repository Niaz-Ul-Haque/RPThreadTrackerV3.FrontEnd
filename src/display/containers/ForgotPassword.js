import React from 'react';
import { Container, Row, Col, Card, CardBlock, Button, Input, InputGroup, InputGroupAddon } from 'reactstrap';

const ForgotPassword = () => (
	<div className="app flex-row align-items-center">
		<Container>
			<Row className="justify-content-center">
				<Col md="6">
					<Card className="mx-4">
						<CardBlock className="p-4">
							<h1>Forgot your password?</h1>
							<p className="text-muted">
								Enter your username or email address below
								and we will email you a replacement password.
							</p>
							<InputGroup className="mb-3">
								<InputGroupAddon><i className="icon-user" /></InputGroupAddon>
								<Input type="text" placeholder="Username/Email" />
							</InputGroup>
							<Button color="primary" className="px-4">Request</Button>
							<span className="pull-right"><a href="/login">Back to Login</a></span>
						</CardBlock>
					</Card>
				</Col>
			</Row>
		</Container>
	</div>
);

export default ForgotPassword;
