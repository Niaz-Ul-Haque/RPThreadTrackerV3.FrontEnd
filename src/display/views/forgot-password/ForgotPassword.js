import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { AvForm } from 'availity-reactstrap-validation';
import { Card, CardBlock, Button, Row, Col } from 'reactstrap';
import { submitUserForgotPassword } from '../../../infrastructure/actions';
import LoadingIndicator from '../../shared/LoadingIndicator';
import ForgotPasswordForm from '../../forms/forgot-password/ForgotPasswordForm';

const propTypes = {
	displayLoadingIndicator: PropTypes.bool.isRequired,
	forgotPasswordError: PropTypes.string,
	submitUserForgotPassword: PropTypes.func.isRequired
};
const defaultProps = {
	forgotPasswordError: ''
};

const mapStateToProps = (state) => {
	const {
		loading,
		errors
	} = state;
	return {
		displayLoadingIndicator: loading.forgotPasswordLoading,
		forgotPasswordError: errors.forgotPasswordError
	};
};

class ForgotPassword extends Component {
	constructor(props) {
		super(props);
		this.handleForgotPasswordSubmit = this.handleForgotPasswordSubmit.bind(this);
		this.handleInputChange = this.handleInputChange.bind(this);
		this.state = {
			forgotPasswordRequest: {}
		};
	}
	handleForgotPasswordSubmit() {
		this.props.submitUserForgotPassword(this.state.forgotPasswordRequest);
	}
	handleInputChange(event) {
		const { target } = event;
		const value = target.type === 'checkbox' ? target.checked : target.value;
		const { name } = target;
		this.setState({
			forgotPasswordRequest: Object.assign({}, this.state.forgotPasswordRequest, {
				[name]: value
			})
		});
	}
	render() {
		const { displayLoadingIndicator, forgotPasswordError } = this.props;
		let loading = (<span />);
		if (displayLoadingIndicator) {
			loading = (
				<LoadingIndicator style={{
					width: 50,
					height: 50,
					position: 'absolute',
					top: 0,
					right: 0
				}}
				/>
			);
		}
		let error = (<span />);
		if (forgotPasswordError) {
			error = (
				<div className="has-danger">
					<p className="form-control-feedback">{forgotPasswordError}</p>
				</div>
			);
		}
		return (
			<Card className="mx-4">
				<CardBlock className="p-4">
					{loading}
					<AvForm onValidSubmit={this.handleForgotPasswordSubmit}>
						<h1>Forgot your password?</h1>
						<p className="text-muted">
							Enter your email address below and we will email you a link to reset your password.
						</p>
						{error}
						<ForgotPasswordForm handleInputChange={this.handleInputChange} />
						<Row>
							<Col xs="6">
								<Button color="primary" className="px-4">Request</Button>
							</Col>
							<Col xs="6" className="text-right text-muted">
								<span className="pull-right"><a href="/login">Back to Login</a></span>

							</Col>
						</Row>
					</AvForm>
				</CardBlock>
			</Card>
		);
	}
}
ForgotPassword.propTypes = propTypes;
ForgotPassword.defaultProps = defaultProps;
export default connect(mapStateToProps, {
	submitUserForgotPassword
})(ForgotPassword);
