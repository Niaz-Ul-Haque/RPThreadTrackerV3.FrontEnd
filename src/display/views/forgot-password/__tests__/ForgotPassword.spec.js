// #region imports
import React from 'react';
import { getSpecWrapper, shallowWithState } from '~/testhelpers/helpers.unit';
import ForgotPassword from '../ForgotPassword';
// #endregion imports

// #region mocks
jest.mock('../../../shared/loading/LoadingIndicator', () => 'LoadingIndicator');
jest.mock('../../../forms/forgot-password/ForgotPasswordForm', () => 'ForgotPasswordForm');
jest.mock('../../../../infrastructure/actions', () => ({}));
// #endregion mocks

const createTestProps = propOverrides => ({
	submitUserForgotPassword: jest.fn(),
	...propOverrides
});

const createTestState = stateOverrides => ({
	loading: { forgotPasswordLoading: false },
	errors: { forgotPasswordError: null },
	...stateOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const state = createTestState();
			const jsx = (<ForgotPassword {...props} />);
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot with loading indicator', () => {
			const props = createTestProps();
			const state = createTestState({ loading: { forgotPasswordLoading: true } });
			const jsx = (<ForgotPassword {...props} />);
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot with error message', () => {
			const props = createTestProps();
			const state = createTestState({ errors: { forgotPasswordError: 'An unknown error occurred.' } });
			const jsx = (<ForgotPassword {...props} />);
			const element = shallowWithState(jsx, state).dive();
			expect(element).toMatchSnapshot();
		});
	});
});

describe('behavior', () => {
	describe('handleForgotPasswordSubmit', () => {
		it('should be called when form is submitted', () => {
			const submitUserForgotPassword = jest.fn();
			const props = createTestProps({ submitUserForgotPassword });
			const state = createTestState();
			const jsx = (<ForgotPassword {...props} />);
			const element = shallowWithState(jsx, state).dive();
			const form = getSpecWrapper(element, 'forgot-password-form-container');
			form.props().onValidSubmit();
			expect(submitUserForgotPassword).toHaveBeenCalledTimes(1);
		});
	});
	describe('handleInputChange', () => {
		it('should handle text field update', () => {
			const event = { target: { type: 'text', name: 'username', value: 'my-username' } };
			const submitUserForgotPassword = jest.fn();
			const props = createTestProps({ submitUserForgotPassword });
			const state = createTestState();
			const jsx = (<ForgotPassword {...props} />);
			const element = shallowWithState(jsx, state).dive();
			element.instance().handleInputChange(event);
			const form = getSpecWrapper(element, 'forgot-password-form-container');
			form.props().onValidSubmit();
			expect(submitUserForgotPassword).toHaveBeenCalledTimes(1);
			expect(submitUserForgotPassword).toHaveBeenLastCalledWith({ username: 'my-username' });
		});
	});
});
