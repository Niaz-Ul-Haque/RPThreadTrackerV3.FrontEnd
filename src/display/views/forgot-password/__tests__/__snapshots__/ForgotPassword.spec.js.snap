// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Card
  className="mx-4"
  tag="div"
>
  <CardBlock
    className="p-4"
    tag="div"
  >
    <span />
    <AvForm
      data-spec="forgot-password-form-container"
      method="get"
      model={Object {}}
      onInvalidSubmit={[Function]}
      onKeyDown={[Function]}
      onSubmit={[Function]}
      onValidSubmit={[Function]}
      tag={[Function]}
      validationEvent={
        Array [
          "onChange",
          "onInput",
        ]
      }
    >
      <h1>
        Forgot your password?
      </h1>
      <p
        className="text-muted"
      >
        Enter your email address below and we will email you a link to reset your password.
      </p>
      <span />
      <ForgotPasswordForm
        handleInputChange={[Function]}
      />
      <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <Button
            className="px-4"
            color="primary"
            tag="button"
          >
            Request
          </Button>
        </Col>
        <Col
          className="text-right text-muted"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <span
            className="pull-right"
          >
            <Link
              href="/login"
              replace={false}
              to="/login"
            >
              Back to Login
            </Link>
          </span>
        </Col>
      </Row>
    </AvForm>
  </CardBlock>
</Card>
`;

exports[`rendering snapshots should render valid snapshot with error message 1`] = `
<Card
  className="mx-4"
  tag="div"
>
  <CardBlock
    className="p-4"
    tag="div"
  >
    <span />
    <AvForm
      data-spec="forgot-password-form-container"
      method="get"
      model={Object {}}
      onInvalidSubmit={[Function]}
      onKeyDown={[Function]}
      onSubmit={[Function]}
      onValidSubmit={[Function]}
      tag={[Function]}
      validationEvent={
        Array [
          "onChange",
          "onInput",
        ]
      }
    >
      <h1>
        Forgot your password?
      </h1>
      <p
        className="text-muted"
      >
        Enter your email address below and we will email you a link to reset your password.
      </p>
      <div
        className="has-danger"
      >
        <p
          className="form-control-feedback"
        >
          An unknown error occurred.
        </p>
      </div>
      <ForgotPasswordForm
        handleInputChange={[Function]}
      />
      <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <Button
            className="px-4"
            color="primary"
            tag="button"
          >
            Request
          </Button>
        </Col>
        <Col
          className="text-right text-muted"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <span
            className="pull-right"
          >
            <Link
              href="/login"
              replace={false}
              to="/login"
            >
              Back to Login
            </Link>
          </span>
        </Col>
      </Row>
    </AvForm>
  </CardBlock>
</Card>
`;

exports[`rendering snapshots should render valid snapshot with loading indicator 1`] = `
<Card
  className="mx-4"
  tag="div"
>
  <CardBlock
    className="p-4"
    tag="div"
  >
    <LoadingIndicator
      style={
        Object {
          "height": 50,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 50,
        }
      }
    />
    <AvForm
      data-spec="forgot-password-form-container"
      method="get"
      model={Object {}}
      onInvalidSubmit={[Function]}
      onKeyDown={[Function]}
      onSubmit={[Function]}
      onValidSubmit={[Function]}
      tag={[Function]}
      validationEvent={
        Array [
          "onChange",
          "onInput",
        ]
      }
    >
      <h1>
        Forgot your password?
      </h1>
      <p
        className="text-muted"
      >
        Enter your email address below and we will email you a link to reset your password.
      </p>
      <span />
      <ForgotPasswordForm
        handleInputChange={[Function]}
      />
      <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <Button
            className="px-4"
            color="primary"
            tag="button"
          >
            Request
          </Button>
        </Col>
        <Col
          className="text-right text-muted"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <span
            className="pull-right"
          >
            <Link
              href="/login"
              replace={false}
              to="/login"
            >
              Back to Login
            </Link>
          </span>
        </Col>
      </Row>
    </AvForm>
  </CardBlock>
</Card>
`;
